### ISSUE TITLE
missing digamma kernel for KnetArray

### ISSUE THREAD

**Issue by CarloLucibello** [denizyuret/Knet.jl#292]
`digamma` is needed if we want to take derivatives of `gamma` and `lgamma`. See also  #290

**Comment by CarloLucibello** [comment]
>Is it possible to adapt this code? https://github.com/rachtsingh/ibp_vae/tree/master/src/lgamma/src

Definitely. I'm no expert here, but I think it could be enough to add a `digamma_impl` here https://github.com/denizyuret/Knet.jl/blob/master/deps/cuda1.jl

and then add to the `unary_ops` list in `unary.jl` the entry
```
("digamma_impl", "digamma", "digamma_impl(xi)"
```

**Comment by xukai92** [comment]
Thanks. How is it possible for me to take derivative of gamma family when not doing broadcasting? Maybe it's on CPU?

**Comment by CarloLucibello** [comment]
On cpu you can compute derivatives even when broadcasting
``` julia
julia> loss(x) = sum(gamma.(x))
loss (generic function with 1 method)

julia> grad(loss)(rand(10))
10-element Array{Float64,1}:
   -9.66163 
   -5.84301 
 -286.216   
  -21.1938  
   -0.907866
   -5.5372  
   -1.23051 
  -16.4395  
   -3.84932 
  -36.5713  
```
